### DISCUSSION TITLE
[Proposal] 用 github 来整理记录社区产生的所有信息

### DISCUSSION THREAD

**Discussion by drafish** [PlanckerLabs/EIP-Research#13] [General]
## 想法的来由

最近有以前的同事问我在做什么。我就大概说了下我在普朗克做公共物品建设。他听了还挺感兴趣，想来找点兼职做。我就把普朗克以前的周会视频发给他让他先看看，然后让他有空可以来参加周会感受下氛围。但我有点怀疑他是否真能找到事情做。

我是带项目入组的，我不需要了解的很清楚，我只要知道普朗克愿意支持我就够了。但我同事不是，他从感兴趣到真找到事情参与进来，这中间会经历很漫长的过程。这个过程没有几个人能坚持下来。

一个新人来 remix cc 看了一圈走了，我会觉得确实没有适合他做的事情。

一个新人来普朗克其他项目看了一圈走了，我会觉得不一定就没有适合他做的事情，可能是因为信息获取成本太高了，他没有耐心去了解，所以走了

所以我才有了这个想法，用 github 来整理记录社区产生的所有信息。

这也不光是为了新人，对我们自己来说也会极大地提升协作效率。而且对融资也有好处。

现在融资小组的小伙伴们去跟投资方讲普朗克在做什么，只能靠嘴说，投资方也只能听他们说。他们不可能把 notion 丢给投资方吧，那太乱了。如果能用 github 把信息都整理清楚，那他们就可以给投资人看 github 。

还有之前周会上一直在讨论的声誉激励问题。之所以声誉激励只能被头部的人所捕获，是因为大家做的事情都不可见，没有记录，或者记录很混乱。

这个问题在 github 上就很好解决。因为所有人做的事情都是有记录的，而且可以很方便地查询到。就比如我们组里的 mocca ，他做了哪些事，看下面这个链接就可以很清楚的了解到

https://github.com/PlanckerLabs/remix-cc/issues?q=is%3Aissue+assignee%3Agongzishuye+

mocca 做了这些贡献，那他就获得了这部分的声誉，他想获得更多的声誉，那就可以去做更多的贡献。他的声誉和他的贡献成正比。

所以用 github 来记录所有信息，这不是为了解决某一个问题，而是可以解决很多问题，有些问题甚至你都没想到，它就已经自行解决了。这是一个全方位的提升，就像你把电脑的机械硬盘换成固态硬盘那样。

## 样品展示

以 Remix CC 为例做个展示

- [wiki](https://github.com/PlanckerLabs/remix-cc/wiki) 用来维护文档类信息，新人进来先看 wiki 就能大概了解我们是什么，在做什么
- [issue](https://github.com/PlanckerLabs/remix-cc/issues) 用来分解任务，并且 assign 到执行人，执行完成就可以 close 掉，整个执行过程都会有记录
- [projects](https://github.com/PlanckerLabs/remix-cc/projects?type=classic) 维护任务面板，哪些在进行中，哪些是已经完成的，哪些是还没开始的，都一目了然
- [milestones](https://github.com/PlanckerLabs/remix-cc/milestones) 维护项目目标并且 link 到具体的任务，可以清楚地看到每个目标包含了哪些任务，完成度是多少
- [discussions](https://github.com/PlanckerLabs/remix-cc/discussions) 用作论坛讨论，如果讨论出可执行的任务，可以直接根据 discussion 生成 issue
- [insights](https://github.com/PlanckerLabs/remix-cc/pulse) 可以查看项目近期的动态，这里面还有个 [traffic](https://github.com/PlanckerLabs/remix-cc/graphs/traffic) 版块专门用来展示项目近期的访问量数据统计

就目前来说，Remix CC 也只是初步完成了信息整理，而且只是我一个人在整理。未来我会要求我的组员有问题尽量在 github 上问。比如对某一个任务有不清楚的点，那就直接在 issue 里面问我。这样他就不需要和我解释上下文，可以直截了当地问出他想问的问题。等任务完成以后还要把产出物写到 issue 里。比如完成了一个视频任务，那就回复一个视频链接；如果给 Remix 官方提交了一个 PR ，那就回复一个 PR 链接。如果任务的执行者希望获得金钱奖励，那可以在 issue 里回复他完成任务花了多少时间，希望获得多少报酬。我会大概做个评估，只要不是很过分，我就可以向普朗克或者 GCC 去申请回溯性奖励。

## 普朗克的 github 设计

按照我的理解，普朗克是一个共创空间，这个空间里有很多项目。这些项目都应该有项目介绍、有 milestones 、有迭代计划、有具体可执行的任务。其实和 Remix CC 并没有什么两样。Remix CC 在用的这套完全可以套用的别的项目上面。

#### [Repositories](https://github.com/orgs/PlanckerLabs/repositories)

每个项目都会有若干个 repo ，但会有一个主 repo 来记录所有信息。普朗克社区也有一个 repo 来记录社区本身的信息，比如运营、融资等等属于社区公共层面的信息。

repo 和 repo 之间的数据都是隔离的。

举个简单的例子，假如我们所有项目组都在现有的论坛上讨论问题，那我们的帖子都会混杂在一起。虽然可以建目录来做区分，但后台数据还是混在一起的。如果我们用 github 的 discussions 来讨论问题，那每个项目组都可以拥有自己独立的论坛。而且这个数据是跟着 repo 走的，而且不光是 discussions，还有 issue、projects、milestones、wiki 等等。如果未来有项目越做越大，要独立出普朗克，那直接建一个新的 github group ，然后把 repo 迁过去就可以了。

#### [Teams](https://github.com/orgs/PlanckerLabs/teams)

每个项目组都创建一个 team ，把组员都加到各自的 team 中。这样每个项目有哪些成员都一目了然了。

我还没用过 Teams ，不知道还有哪些神奇的功能。未来在使用过程中再慢慢发掘吧。

#### [People](https://github.com/orgs/PlanckerLabs/people)

目前普朗克的 github group 中 Members 是 21 个，Outside collaborators 是 10 个。

但什么人应该是 Members ，什么人应该是 Outside collaborators ，目前并没有很清晰的界定。我建议是把 CBI 成员定义为 Members ，Bounty 成员定义为 Outside collaborators 。

出现在这份名单里面的人都是会获得声誉激励的，而且 Members 的声誉激励会更大。

以太坊官方的 github group 有 92 个 Members ，我不需要了解每个人具体做了什么，他只要出现在这份名单里，我就觉得他很牛。我希望普朗克的这份名单也能起到同样的效果。

**Comment by brucexu-eth** [comment]
主要是不是所有人都是程序员。然后要看具体是什么目的吧，我看到有两个：

1. 成员了解其他项目组的动态和进展。这个没什么好办法，只能整理的好一点，周会同步或者自己项目组开会，感兴趣的参加。GitHub 上也是一样的问题，也不保证能不能看到。
2. 新成员快速了解能做什么事情。LXDAO 是独立设计了一个 bounty task，放一些适合新人来做的任务，也可以有钱赚。有人问就发给他。但是普朗克的策略是专注大项目和方向深度探索。这种背景下，其实天然就有进入的门槛，因此确实需要多参加一些会，或者定向的 onboarding 才可以。

论坛利用率确实不高，LXDAO 的其中一个解法是小组的周会也可以在论坛创建帖子，大家的会议讨论视频、评论等放上去，这样能增加点活跃度。这是 LXDAO 周会的帖子 https://forum.lxdao.io/t/lxdao-community-call-59/864 对比的话，普朗克是在 notion 上面。

**Reply by drafish** [reply]
我倒觉得是不是程序员不是很重要。github 也不是程序员的专属。只要不涉及代码提交，pull request 这些操作，其他操作应该适用于所有人。

**Comment by York-Aaron-Swartz** [comment]
个人还是很认同，用github的

**Comment by ethz42** [comment]
个人认同
